cmd/renterd/config.go

```diff
@@ -674,6 +674,7 @@ func setStoreConfig(cfg *config.Config) {
 
 func setS3Config(cfg *config.Config) {
 	if !promptYesNo("Would you like to configure S3 settings?") {
+		cfg.S3.Enabled = promptYesNo("Would you like to enable the S3 gateway?")
```

Comment:
I'd move this below the "Would you like to configure S3 settings" check. If the user doesn't want to configure the settings at all we just return without modifying the defaults. Then we ask if they want to enable it. If they don't want to enable it, we return with Enabled = false. If they want to enable it, we move on to the remaining prompts.